Command/AbstractCommand.php

```diff
@@ -270,6 +270,14 @@ protected function getConnectionData($name)
     protected function getDsn($connectionName)
     {
         $connection = $this->getConnectionData($connectionName);
+        // Add user and password to dsn string
+        $dsn = explode(';', $connection['dsn']);
+        if (isset($connection['user'])) {
+            $dsn[] = 'user=' . urlencode($connection['user']);
+        }
+        if (isset($connection['password'])) {
+            $dsn[] = 'password=' . urlencode($connection['password']);
+        }
 
         return $connection['dsn'];
```

Comment:
That's the wrong place to modify the DSN. The DSN is vendor specific, so it should be set correctly when read here.